js/map.js

```diff
@@ -188,23 +219,135 @@ window.map = (function () {
     };
     isPageActive = false;
 
-    var mapPins = document.querySelectorAll('.map__pin:not(.map__pin--main)');
     mapPinsMainElement.style.top = MAP_PIN_TOP + 'px';
     mapPinsMainElement.style.left = MAP_PIN_LEFT + 'px';
     mapElement.classList.add('map--faded');
     adFormContainer.classList.add('ad-form--disabled');
+    filterTypeElement.value = 'any';
+    filterPriceElement.value = 'any';
+    filterRoomsElement.value = 'any';
+    filterGuestsElement.value = 'any';
+    
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        elem.checked = false;
+      }
+    });
+    
+    removeCard();
+    removePins();
+  }
 
-    if (mapElement.querySelector('.popup')) {
-      document.querySelector('.popup__close').removeEventListener('click', closeCardHandler);
-      document.removeEventListener('keydown', escPressCloseCardHandler);
-      removeCard();
-    }
+  function filter() {
+    var type = filterTypeElement.value;
+    var price = filterPriceElement.value;
+    var rooms = filterRoomsElement.value;
+    var guests = filterGuestsElement.value;
+    var featuresArray = [];
 
-    Array.prototype.forEach.call(mapPins, function (elem) {
-      elem.parentElement.removeChild(elem);
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        featuresArray.push(elem.value);
+      }
     });
+
+    var filteredPropertyType;
+    var filteredPropertyPrice;
+    var filteredPropertyRooms;
+    var filteredPropertyGuests;
+    var filteredPropertyFeatures;
```

Comment:
Reduce the amount of this variable to one! 
One `result` variable.

Reply:
Done